server/main.py

```diff
@@ -129,7 +129,9 @@ async def index(request):
         job.command(f'git merge-base --is-ancestor {_shell_escape(commit)} HEAD')
         job.command(f'git checkout {_shell_escape(commit)}')
         # Make sure the file is in the repository.
-        script_file = script.partition(' ')[0]
+        if not isinstance(script, list):
```

Comment:
Should we really support different input types here? This will later be converted into a single string anyhow, so maybe we should just pass a string from the CLI too? Supporting a list here complicates the logic below where we split by spaces for the escaping.

Reply:
The current implementation won't support parameters with spaces in it, eg: "this is a single param with spaces". We get an array straight from the argument parser, and it's more convenient for the server to have an array anyway based on how arguments are quoted.

Reply:
Won't you have the same problem though once you assemble the command for Hail Batch to run? I.e. `Job.command` doesn't support a list either.

Reply:
The original code splits by the space character, and then shell quotes each element. I've just moved this up higher, and added support for if you supply the array directly:

Original:
https://github.com/populationgenomics/analysis-runner/blob/4ea28daeab2c5480749024c1e1cae5f60f225ab9/server/main.py#L158-L159

Reply:
Got it, thanks!